serial.c

```diff
@@ -124,7 +124,7 @@ uint8_t serial_read()
     uint8_t data = rx_buffer[tail];
 
     tail++;
-    if (tail == (uint8_t)RX_BUFFER_SIZE) { tail = 0; }
+    if (tail == RX_BUFFER_SIZE) { tail = 0; }
```

Comment:
Because it was squelching a valid error.  Tail could never be equal to RX_BUFFER_SIZE (256).  If we tried to increase this further, it would just silently work incorrectly.